tests/src/test/spark330/scala/com/nvidia/spark/rapids/BloomFilterAggregateQuerySuite.scala

```diff
@@ -65,74 +68,149 @@ class BloomFilterAggregateQuerySuite extends SparkQueryCompareTestSuite {
         (1L to 100L).map(_ => None)).toDF("col")
   }
 
-  for (numEstimated <- Seq(4096L, 4194304L, Long.MaxValue,
+  private def withExposedSqlFuncs[T](spark: SparkSession)(func: SparkSession => T): T = {
+    try {
+      installSqlFuncs(spark)
+      func(spark)
+    } finally {
+      uninstallSqlFuncs(spark)
+    }
+  }
+
+  private def doBloomFilterTest(numEstimated: Long, numBits: Long): DataFrame => DataFrame = {
+    df =>
+      val table = "bloom_filter_test"
+      val sqlString =
+        s"""
+           |SELECT might_contain(
+           |            (SELECT bloom_filter_agg(col,
+           |              cast($numEstimated as long),
+           |              cast($numBits as long))
+           |             FROM $table),
+           |            col) positive_membership_test,
+           |       might_contain(
+           |            (SELECT bloom_filter_agg(col,
+           |              cast($numEstimated as long),
+           |              cast($numBits as long))
+           |             FROM values (-1L), (100001L), (20000L) as t(col)),
+           |            col) negative_membership_test
+           |FROM $table
+          """.stripMargin
+      df.createOrReplaceTempView(table)
+      withExposedSqlFuncs(df.sparkSession) { spark =>
+        spark.sql(sqlString)
+      }
+  }
+
+  private def getPlanValidator(exec: String): (SparkPlan, SparkPlan) => Unit = {
+    def searchPlan(p: SparkPlan): Boolean = {
+      if (ExecutionPlanCaptureCallback.didFallBack(p, exec)) {
+        true
+      } else if (p.children.exists(searchPlan)) {
+        true
+      } else if (p.subqueries.exists(searchPlan)) {
+        true
+      } else {
+        false
+      }
+    }
+    (_, gpuPlan) => {
+      val executedPlan = ExecutionPlanCaptureCallback.extractExecutedPlan(gpuPlan)
+      assert(searchPlan(executedPlan), s"Could not find $exec in the GPU plan:\n$executedPlan")
+    }
+  }
+
+  // test with GPU bloom build, GPU bloom probe
+  for (numEstimated <- Seq(4096L, 4194304L,
     SQLConf.RUNTIME_BLOOM_FILTER_MAX_NUM_ITEMS.defaultValue.get)) {
-    for (numBits <- Seq(4096L, 4194304L, Long.MaxValue,
+    for (numBits <- Seq(4096L, 4194304L,
```

Comment:
before we would check `Long.MaxValue` here in addition to `RUNTIME_BLOOM_FILTER_MAX_NUM_BITS.defaultValue`. I am mostly on curious on why we made the change, but I do wonder if we should go above `defaultValue`?

Reply:
I pulled out Long.MaxValue here primarily to save on test time.  These tests take a while as it is, and it adds literally minutes to add the extra Long.MaxValue cases.  Happy to add them back if we think it's worth it.

Reply:
Ok that makes sense to me. I am fine without it, thanks for the feedback